blogs/dns_tracking/dns_tracking.md

```diff
@@ -0,0 +1,136 @@
+eBPF (extended Berkeley Packet Filter).
+In this blog post, we'll delve into the world of DNS tracking using eBPF
+tracepoint hooks, exploring how this powerful combination can be used for
+various purposes, including network monitoring and security enhancement.
+
+## Understanding DNS Resolution
+
+Before diving into the specifics of eBPF tracepoint hooks, let's briefly
+recap how DNS resolution works.
+When you enter a website's domain name (e.g., www.example.com) in your
+browser, your computer needs to find the corresponding IP address.
+This process involves multiple steps, including querying DNS servers,
+caching responses, obtaining the IP address to establish a connection
+and caching response for future re-occurrence of same DNS query.
+
+## Utilizing Tracepoint Hooks for DNS Tracking
+
+Tracepoint hooks are predefined points in the Linux kernel where eBPF
+programs can be attached to capture and analyze specific events.
+For DNS tracking, we leveraged tracepoint hooks associated with DNS
+resolution processes specifically `tracepoint/net/net_dev_queue` tracepoint,
+Then we parse the DNS header to determine if its query or response, attempt
+to correlate query and response for specific DNS transaction and then record the
+elapsed time to compute DNS latency. Further, DNS network flows are enriched
+to include fields, viz. DNS Id, DNS latency and DNS response codes to help
+build graphs with aggregated DNS statistics and to help filtering on specific
+fields in Network Observability console.
+
+## Potential Use Cases
+
+DNS tracking with eBPF tracepoint hooks can serve various purposes:
+
+- Network Monitoring: Gain insights into DNS queries and responses,
+helping network administrators identify unusual patterns,
+potential bottlenecks, or performance issues.
+
+- Security Analysis: Detect suspicious DNS activities, such as domain
+name generation algorithms (DGA) used by malware,
+or identify unauthorized DNS resolutions that might indicate a security breach.
+
+- Troubleshooting: Debug DNS-related issues by tracing DNS resolution steps,
+tracking latency, and identifying misconfigurations.
+
+## How to enable DNS tracking
+
+By default DNS tracking is disabled because it requires
+`privileged` access, to enable this feature we need to create a flow
+collector object with the following fields enabled in eBPF config
+section
+
+```yaml
+apiVersion: flows.netobserv.io/v1beta1
+kind: FlowCollector
+metadata:
+  name: cluster
+spec:
+  agent:
+    type: EBPF
+    ebpf:
+      privileged: true
+      features:
+        - DnsTacking
+```
+
+## A quick tour in the UI
+
+Once `DnsTacking` feature enabled, the Console plugin will automatically
+adapt to provide additional filters and show informations across views.
+
+Open your OCP Console and move to
+`Administrator view` -> `Observe` -> `Network Traffic` page as usual.
+
+Three new filters, `DNS Id`, `DNS Latency` and `DNS Response Code` will be available
+in the common section:
```

Comment:
```suggestion
Three new filters, `DNS Id`, `DNS Latency` and `DNS Response Code` are available
in the common section:
```

Reply:
Just a suggestion, but present tense is more appropriate for procedural information than future tense.